runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -76,13 +80,24 @@ class ExecutionContext final {
     std::unique_ptr<QuantumDevice> _driver_ptr{nullptr};
     std::unique_ptr<MemoryManager> _driver_mm_ptr{nullptr};
 
+#ifdef __device_openqasm
+    std::unique_ptr<Device::OpenQasm::PythonInterpreterGuard> _py_guard{nullptr};
+#endif
+
```

Comment:
Is there a way around these pre-processor statements?
I think the fact that the runtime was compiled with OQ support should not imply that we always initialize a Python interpreter instance, even if the device is never used.